cdhweb/pages/management/commands/exodus.py

```diff
@@ -274,99 +253,46 @@ def migrate_pages(self, page, parent):
+                # if the files are the same
+                if samefile:
+                    # keep the first one and remove the others from the
+                    # list of files to be migrated
+                    extra_copies = val[1:]
 
-    def profile_pages(self):
-        """Exodize all existing Profiles to new Profile model."""
-        # all the fields on Profile that moved to Person will have been handled
-        # by django migrations; we just need to create new Page models
-        people_landing = PeopleLandingPage.objects.first()
-        if not people_landing:
-            return
-        # only create profiles if there's actually text in the old one
-        for profile in OldProfile.objects.exclude(bio=""):
-            person = Person.objects.get(user=profile.user)
-            profile_page = Profile(
-                person=person,
-                title=profile.title,
-                slug=self.convert_slug(profile.slug),
-                image=self.get_wagtail_image(profile.image) if profile.image else self.get_wagtail_image(
-                    profile.thumb) if profile.thumb else None,
-                education=profile.education,
-                bio=json.dumps([
-                    {"type": "migrated", "value": profile.bio},
-                ])
-            )
-            # added as child of people landing page so slugs are correct
-            people_landing.add_child(instance=profile_page)
-            people_landing.save()
-            # if the old profile wasn't published, unpublish the new one
-            if profile.status != CONTENT_STATUS_PUBLISHED:
-                profile_page.unpublish()
+                # if not all the same, identify the largest
+                # (all are variant/cropped versions of the same image)
+                else:
+                    largest_file = None
+                    largest_size = 0
+                    for filepath in val:
+                        size = os.stat(filepath).st_size
+                        if size > largest_size:
+                            largest_size = size
+                            largest_file = filepath
 
-    def get_collection(self, name):
-        # if we don't already have this collection, get it
-        if name not in self.collections:
-            # try to get it if it already exists
-            coll = Collection.objects.filter(name=name).first()
-            # otherwise, create it
-            if not coll:
-                coll = Collection(name=name)
-                self.collections['root'].add_child(instance=coll)
-                self.collections['root'].save()
+                    extra_copies = [f for f in val if f != largest_file]
 
-            self.collections[name] = coll
+                # remove duplicate and variant images that
+                # will not be imported into wagtail
+                for extra_copy in extra_copies:
+                    filenames.remove(extra_copy)
 
```

Comment:
oh, nice – I've been wondering about how/when we would clean up the images we aren't actually using

Reply:
i think this also is your code

Reply:
Ha, ok. Then it probably doesn't do what I hoped it does! 😆